py/makemanifest.py

```diff
@@ -0,0 +1,56 @@
+import sys
+import os
+import subprocess
+
+assert sys.argv[1] == '-o'
+output_file = sys.argv[2]
+
+TOP = sys.argv[3]
+BUILD = sys.argv[4]
+mpy_cross_flags = sys.argv[5]
+manifests = sys.argv[6:]
+
+MPY_CROSS = TOP + '/mpy-cross/mpy-cross'
+MPY_TOOL = TOP + '/tools/mpy-tool.py'
+
+frozen_list = []
+
+def system(cmd):
+    try:
+        p = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        output = p.stdout
+        return 0, output
+    except subprocess.CalledProcessError as er:
+        err = er
+        return -1, err
+
+def mkdir(path):
+    cur_path = ''
+    for p in path.split('/')[:-1]:
+        cur_path += p + '/'
+        try:
+            os.mkdir(cur_path)
```

Comment:
This function could be replaced by `os.makedirs(path, exist_ok=True)`, unless python2 support is still desired?
https://docs.python.org/3.5/library/os.html#os.makedirs

Reply:
Thanks, I missed that.

But I do think it's important to have Python 2 support, since all the other scripts have it and on some embedded systems Python 2 is all they have.

Reply:
I would hope we really really wont be writing any Python2 code ever again. Its dead!

Reply:
I do agree with Damien here, there's plenty of 'stable' systems that are Python2 still. I personally won't be writing ugly code again for the sake of compatibility, but this here is a particularly minor concession.

Reply:
Why does "What embedded systems have" impact the build process at all?

Reply:
And for what it is worth, this `makemanifest.py` file is not Python2 compatible.

Reply:
> Why does "What embedded systems have" impact the build process at all?

Eg building MPY on an old RPi or similar SoM running Linux, or in a virtualbox image that is old and you can't update for some reason.